pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -351,6 +352,11 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
           ? TableNameBuilder.OFFLINE.tableNameWithType(rawTableName)
           : TableNameBuilder.REALTIME.tableNameWithType(rawTableName);
 
+      if (UploadedRealtimeSegmentName.isUploadedRealtimeSegmentName(segmentName) && tableType != TableType.REALTIME) {
+        throw new ControllerApplicationException(LOGGER,
+            "isUploadedRealtimeSegmentName is only allowed for realtime table", Response.Status.BAD_REQUEST);
```

Comment:
This message is not user friendly. Suggest something like `Cannot upload segment: <segmentName> to OFFLINE table as this format is reserved for uploaded real-time segment`